TreeDB/internal/vectorpartition/vectorpartition_test.go

```
@@ -0,0 +1,596 @@
+	}
+	if _, err := DecodeArtifact(raw, len(raw)); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := DecodeArtifact(raw, len(raw)-1); err == nil {
+		t.Fatal("over-max decoder input accepted")
+	}
+	if _, err := DecodeArtifact(append(raw, '\n'), len(raw)+1); err == nil {
+		t.Fatal("noncanonical whitespace accepted")
+	}
+	a.Metrics.EdgeCut++
+	bad, _ := json.Marshal(a)
```

golangci-lint (2.12.2)

[error] Error return value of `encoding/json.Marshal` is not checked: unsafe type `float64` found

(errchkjson)